pkg/cloud/azure/services/internalloadbalancers/internalloadbalancers.go

```diff
@@ -43,7 +43,7 @@ func (s *Service) Get(ctx context.Context, spec azure.Spec) (interface{}, error)
 		return network.LoadBalancer{}, errors.New("invalid internal load balancer specification")
 	}
 	//lbName := fmt.Sprintf("%s-api-internallb", s.Scope.Cluster.Name)
-	lb, err := s.Client.Get(ctx, s.Scope.ClusterConfig.ResourceGroup, internalLBSpec.Name, "")
+	lb, err := s.Client.Get(ctx, s.Scope.ResourceGroup().Name, internalLBSpec.Name, "")
```

Comment:
If resource group is a configuration, please move this to spec for each service using it instead of this

Reply:
See my comments below. We should be leveraging the defined API types as opposed to a package-level Spec.